tools/importPlanGoals.js

```diff
@@ -0,0 +1,171 @@
+import { readFileSync } from 'fs';
+import parse from 'csv-parse/lib/sync';
+import { option } from 'yargs';
+import {
+  Role, Topic, RoleTopic, Goal, TopicGoal, Grantee, Ttaplan,
+} from '../src/models';
+
+let grantees;
+
+function getItemId(array, item) {
+  if (!array.includes(item)) {
+    array.push(item);
+  }
+  const itemId = array.indexOf(item) + 1;
+  return itemId;
+}
+
+/**
+ * Processes data from .csv creating data arrays and then inserting it to the database
+ */
+async function importGoals() {
+  try {
+    const cleanRoles = [];
+    const cleanTopics = [];
+    const cleanRoleTopics = [];
+    const cleanGoals = [];
+    const cleanGrantees = [];
+    const cleanTtaPlans = [];
+    const cleanTopicGoals = [];
+
+    await grantees.forEach((el, inx) => {
+      let currentGrantee;
+      let currentGranteeId;
+      let grants;
+      let currentGrants = [];
+      let currentGoalName;
+      let currentGoal = {};
+      let currentGoalId;
+      const currentGoals = [];
+
+      Object.keys(el).forEach((key) => {
+        if (key && key.startsWith('Grantee (distinct')) {
+          currentGrantee = el[key] ? el[key].split('|')[0] : `Unknown Grantee ${inx}`;
+          currentGranteeId = getItemId(cleanGrantees, currentGrantee.trim());
+          grants = el[key] ? el[key].split('|')[1] : `Unknown Grant ${inx}, Unknown Grant ${inx + 10000}`;
```

Comment:
Why the two unknown grants?

Reply:
I thought the "Grantee" column would be in this format "Grantee Name | Grant1, Grant2" or "Grantee Name | Grant1". In other words we couldn't guarantee that there is always one grant following the grantee name. Is that not correct?